assets/js/components/projects/ProjectSettings.jsx

```diff
@@ -0,0 +1,366 @@
+      initialSuccessRate,
+      eligibilityRate,
+      responseRate,
+      validRespondentRate,
+      detailedRates,
+      archiveAction,
+      readOnly,
+    } = this.state
+
+    const { t } = this.props
+    const fieldsToSave = [
+      "name",
+      "timezone",
+      "colourScheme",
+      "initialSuccessRate",
+      "eligibilityRate",
+      "validRespondentRate",
+      "detailedRates",
+    ]
+    const errors = {} //this.isNew() ? {} : this.props.errors
+
+    const setTimezone = (timezone) => this.setState({ timezone: timezone })
+
+    const inputProjectName = (
+      <div>
+        <label className="gray-text">Name</label>
+        <input type="text" value={name} />
+      </div>
+    )
+
+    const inputTimeZone = (
+      <div>
+        <TimezoneAutocomplete selectedTz={timezone} readOnly={readOnly} onChange={setTimezone} />
+      </div>
+    )
+
+    const inputColourScheme = (
+      <div>
+        <div className="card-content">
+          <div className="row">
+            <label className="gray-text">{t("Select color scheme")}</label>
+          </div>
+          <div className="row">
+            <div className="col s12">
+              <input
+                id={`defaultScheme`}
+                type="radio"
+                name="toggleDefault"
+                checked={colourScheme == "default"}
+                disabled={readOnly}
+                onChange={(e) => this.setState({ colourScheme: "default" })}
+                className="with-gap"
+              />
+              <label className="colourScheme" htmlFor={`defaultScheme`}>
+                {t("Default color scheme")}
+              </label>
+            </div>
+          </div>
+          <div className="row">
+            <div className="col s12">
+              <input
+                id={`betterDataForHealthScheme`}
+                type="radio"
+                name="toggleDefault"
+                checked={colourScheme == "better_data_for_health"}
+                disabled={readOnly}
+                onChange={(e) => this.setState({ colourScheme: "better_data_for_health" })}
+                className="with-gap"
+              />
+              <label className="colourScheme" htmlFor={`betterDataForHealthScheme`}>
+                {t("Data for health initiative")}
+              </label>
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+
+    const inputInitialSuccessRate = (
+      <div>
+        <div className="row">
+          <div className="col s3" id="initialSuccessRate">
+            <label className="gray-text">Initial success rate</label>
+            <input
+              type="number"
+              step="0.01"
+              min="0"
+              max="1"
+              value={initialSuccessRate}
+              disabled={readOnly}
+              onInput={(e) => this.setState({ initialSuccessRate: e.target.value })}
+              onChange={(e) => this.setState({ initialSuccessRate: e.target.value })}
```

Comment:
You can parse as float immediately, so you state will already contains floats (not strings):
```js
this.setState({ initialSuccessRate: parseFloat(e.target.value, 10) })
```

Also specifying base 10 never hurts.

Reply:
This actually doesn't work: we need to keep the actual value as a String otherwise we can't type `0.01` for example.

Sigh, React :face_exhaling: